src/Components/NavigationBar/NavigationBar.tsx

```diff
@@ -1,18 +1,64 @@
+import { docsNavSignal, navSignal } from '../../Store/navigation-bar-store'
+import { createSignal } from 'solid-js';
+import { Show } from 'solid-js'
 
 export const NavigationBar: Component = () => {
-    const [navs, _] = navSignal;
-
+    const [showDocumentationItems, setShowDocumentationItems] = createSignal(false);
+    const [showMobileNav, setShowMobileNav] = createSignal(false);
+    const [navs] = navSignal;
+    const [docsNav] = docsNavSignal;
+    const docs_nav = docsNav().map(item => <NavLink activeClass="before:mobile-nav-after" class="flex relative justify-start items-center py-2.5 pr-4 pl-2 mb-1.5 text-sm" href={item.path} end>{item.label}</NavLink>)
     return (
-      <header>
-        <div class="w-full h-[56px] bg-[#0e0e0e] text-white">
-          <nav id="navbar" class="flex-container navbar">
-            {navs().map(n => <NavLink href={n.path}>{n.label}</NavLink>)}
-          </nav>
+        <div class='bg-[#0e0e0e] w-full flex flex-col desktop:items-center '>
+            <div class='inner-shadow w-full h-8 flex items-center mobile:hidden'>
+                <div class='text-white-gray w-4/5 text-xs my-0 mx-auto'>
+                    <a class='pr-10' href="https://deriv.com">Deriv website</a>
+                    <a class='pr-10' href='https://deriv.com/about'>About us</a>
+                    <a class='pr-10' href="https://deriv.com/contact-us">Contact us</a>
+                </div>
+            </div>
+            <header class='h-[73px] w-4/5 mobile:h-[48px] mobile:pl-4 mobile:w-full mobile:flex'>
+                <section class="flex items-center">
+                    <img class="desktop:hidden mobile:mr-2.5" src={`/src/assets/${showMobileNav() ? "close.svg" : "hamburger_menu.svg"}`} alt="Deriv Logo" width="16" height="16" onclick={() => setShowMobileNav(!showMobileNav())} />
+                    <NavLink class="flex items-center text-white" href="/">
+                        <img src='/src/assets/deriv.svg' class="h-[73px] w-[180px] mobile:h-auto mobile:w-[84px]" height='73' width="180" alt="Deriv Logo" />
+                        <h1 class='inline ml-6 text-2xl  mobile:text-base mobile:ml-2'>API</h1>
+                    </NavLink>
+                    <nav class="flex flex-row ml-5% mobile:hidden">
+                        {navs().map(n => <NavLink activeClass="after:active-page after:navigation-transition" href={n.path} class="relative text-base mr-10 text-white before:active-page hover:before:navigation-transition" end={n.path === "/"}>{n.label}</NavLink>)}
+                    </nav>
+                </section>
+            </header>
+            <section class={`fixed bg-white top-[48px] h-[calc(100%+54px) w-[253px] overflow-hidden left-[-254px] text-base	burger ${showMobileNav() && "transition-left"}`}>
+                <div class="flex flex-col h-full py-6 pr-4 pl-5">
+                    <div class="menu-wrapper">
+                        {navs().map(n => {
```

Comment:
I wonder if it can work reactively in Solid. I mean there are helper components introduced such as `<For>` https://www.solidjs.com/docs/latest/api#control-flow ```Solid uses components for control flow. The reason is that with reactivity to be performant we have to control how elements are created. For example with lists, a simple map is inefficient as it always maps everything. This means helper functions.``` Can you try `<For />` helper component? Just as you used `<Show />` later?

Reply:
Yep, You are right! I changed all of them to For and also Show
We have to consider that I'm coming from React culture :-D